cmd/yggd/main.go

```diff
@@ -140,6 +140,13 @@ func setupClient(
 		if err != nil {
 			return nil, nil, cli.Exit(fmt.Errorf("cannot create HTTP transport: %w", err), 1)
 		}
+	case "none":
+		var err error
+		transporter, err = transport.NewNoopTransport()
+		if err != nil {
+			return nil, nil, cli.Exit(fmt.Errorf("cannot create no-op transport: %w", err), 1)
+		}
+		log.Info("no network protocol specified; no data will be sent or received over the network")
```

Comment:
When I started `yggd` with `--server` option, then this option was ignored as you can see:

```
[jhnidek@thinkpad-p1 yggdrasil]$ ./builddir/cmd/yggd/yggd --protocol none --server tcp://localhost:1883 --log-level trace --client-id $(hostname)
[yggd] 2023/08/14 10:39:33 /home/jhnidek/github/redhat-insights/yggdrasil/cmd/yggd/main.go:317: starting yggd version 0.3.2
[yggd] 2023/08/14 10:39:33 /home/jhnidek/github/redhat-insights/yggdrasil/cmd/yggd/main.go:149: no network protocol specified; no data will be sent or received over the network
[yggd] 2023/08/14 10:39:33 /home/jhnidek/github/redhat-insights/yggdrasil/internal/work/dispatcher.go:65: connecting to session bus for worker IPC: unix:path=/run/user/1000/bus
[yggd] 2023/08/14 10:39:33 /home/jhnidek/github/redhat-insights/yggdrasil/internal/work/dispatcher.go:92: exported /com/redhat/Yggdrasil1/Dispatcher1 on bus
[yggd] 2023/08/14 10:39:33 /home/jhnidek/github/redhat-insights/yggdrasil/cmd/yggd/client.go:149: connecting to session bus: unix:path=/run/user/1000/bus
[yggd] 2023/08/14 10:39:33 /home/jhnidek/github/redhat-insights/yggdrasil/cmd/yggd/client.go:174: exported /com/redhat/Yggdrasil1 on bus
[yggd] 2023/08/14 10:39:33 /home/jhnidek/github/redhat-insights/yggdrasil/cmd/yggd/main.go:253: no TLS configuration, disabling TLS watcher update
```

I think that it should be good to add there some warning/info message that the `--server` option will be ignored, when `--protoco none` is used. Something like this:

```go
if len(config.DefaultConfig.Server) > 0 {
    log.Infof("server option: %v is ignored for no network protocol", config.DefaultConfig.Server[0]
}
```

Reply:
This starts to get into an area we need to be very cautious. Once flags start becoming mutually exclusive with each other, we begin creating a complicated matrix of flag compatibility. I do not want to maintain a "validateOptions" method that will only grow in complexity as the software matures.

I like your approach though. Rather than stopping the execution of the program with an option validation error, logging a warning that the server value will be ignored can very easily be added in the block of code that sets up the transport.